### PR TITLE
kv/union_store:move implementation of unionstore into store/tikv

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?
This PR moves the implementation of `unionstore` from kv to tikv 
Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/kv/keyflags.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package kv
```
Comment (disksing) [inline store/tikv/kv/keyflags.go:1]: license missing.

File: store/tikv/unionstore/interface.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package unionstore
```
Comment (disksing) [inline store/tikv/unionstore/interface.go:1]: license missing

File: store/tikv/unionstore/union_store.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
+package unionstore
```
Comment (disksing) [inline store/tikv/unionstore/union_store.go:1]: license missing

**Review by disksing — COMMENTED** [review]
rest LGTM

**Review by disksing — APPROVED** [review]

**Review by andylokandy — APPROVED** [review]

**Pull request merged**